### PR TITLE
Adding ability to pass callback functions as props

### PR DESCRIPTION
**Pull request by S33G**
I wanted to use this component, and sync an image changing with the text. So I worked out how to add callback functions.

This can be used like this:
```
  <ReactRotatingText
      items={[
          'This is a really long string for testing',
          'More long strings',
          'Wow this is a even bigger string',
          'Listeners',
      ]}
      onDeletingStart={function() {
          console.log('deleting start');
      }}
      onDeletingEnd={function() {
          console.log('deleting end');
      }}
      onTypingStart={function() {
          console.log('typing start')
      }}
      onTypingEnd={function() {
          console.log('typing end')
      }}
  />
```

### CODE REVIEW

File: src/ReactRotatingText.js [hunk 1]
```diff
@@ -47,6 +47,9 @@ class ReactRotatingText extends React.Component {
     if (output.length < text.length) {
       this._loop(loopingFunc, typingInterval);
     } else {
+	  if (typeof this.props.onTypingEnd == "function") {
+	  	this.props.onTypingEnd(this);
+	  }
```
Comment (adrianmcli) [inline src/ReactRotatingText.js]: Looks like some inconsistent formatting, can you run `eslint --fix` to correct this?

File: src/ReactRotatingText.js [hunk 2]
```diff
@@ -47,6 +47,9 @@ class ReactRotatingText extends React.Component {
     if (output.length < text.length) {
       this._loop(loopingFunc, typingInterval);
     } else {
+	  if (typeof this.props.onTypingEnd == "function") {
+	  	this.props.onTypingEnd(this);
```
Comment (adrianmcli) [inline src/ReactRotatingText.js]: Not sure why you are passing `this` into the function here.

File: src/ReactRotatingText.js [hunk 3]
```diff
@@ -56,13 +59,19 @@ class ReactRotatingText extends React.Component {
     const { deletingInterval } = this.props;
     const loopingFunc = this._erase.bind(this, callback);
 
+    if (typeof this.props.onDeletingStart == "function") {
+	  this.props.onDeletingStart(this);
```
Comment (adrianmcli) [inline src/ReactRotatingText.js]: Same as above, no need to pass in `this`.

File: src/ReactRotatingText.js [hunk 4]
```diff
@@ -56,13 +59,19 @@ class ReactRotatingText extends React.Component {
...
+    }
     // set the string one character shorter
     this.setState({output: output.substr(0, output.length - 1)});
 
     // if we're still not done, recursively loop again
     if (output.length !== 0) {
       this._loop(loopingFunc, deletingInterval);
     } else {
+	  if (typeof this.props.onDeletingEnd == "function") {
+		this.props.onDeletingEnd(this);
```
Comment (adrianmcli) [inline src/ReactRotatingText.js]: Same as above, no need to pass in `this`.

File: src/ReactRotatingText.js [hunk 5]
```diff
@@ -105,7 +114,11 @@ class ReactRotatingText extends React.Component {
     const nextWord = () => {
       this.setState({index: nextIndex});
       this._loop(loopingFunc, emptyPause);
-    };
+	};
+
+    if (typeof this.props.onTypingStart == "function") {
+	  this.props.onTypingStart(this);
```
Comment (adrianmcli) [inline src/ReactRotatingText.js]: Same as above, no need to pass in `this`.

**Review by adrianmcli — CHANGES_REQUESTED** [review]

File: src/ReactRotatingText.js (hunk 1 above)
Comment (charj) [reply src/ReactRotatingText.js]: In my use case, I want to refer to `state.index` so I can select the image which matches the text. Do you have a better way of achieving this?

**Comment by charj** [comment]
Updated @adrianmcli

**Comment by charj** [comment]
Any further thoughts on this? I'm about to push a site to production with this code and don't want to host it from my personal repo ideally

**Comment by charj** [comment]
We're live!
https://getchip.uk/

**Comment by adrianmcli** [comment]
Sorry! This slipped my mind.

**Pull request merged**